python_modules/dagster-webserver/dagster_webserver/webserver.py

```diff
@@ -231,15 +230,31 @@ def index_html_endpoint(self, request: Request):
                 """)
 
     def build_static_routes(self):
-        def next_file_response(file_path):
-            with open(file_path, encoding="utf8") as f:
-                content = f.read().replace(
-                    "BUILDTIME_ASSETPREFIX_REPLACE_ME", f"{self._app_path_prefix}"
-                )
-                return Response(content=content, media_type=guess_type(file_path)[0])
+        temp_dir = tempfile.TemporaryDirectory() if self._uses_app_path_prefix else None
+
+        def next_file_response(path, file_path):
+            # Create the full temporary directory path, including subdirectories
+            temp_dir_path = os.path.join(temp_dir.name, *os.path.dirname(path).split(os.path.sep))
+
+            # Ensure the temporary directory and its parent directories exist
+            os.makedirs(temp_dir_path, exist_ok=True)
+
+            # Create the full path to the temporary file
+            temp_file_path = os.path.join(temp_dir_path, os.path.basename(path))
+
+            if not os.path.exists(temp_file_path):
+                # If the temporary file does not exist, create it and write the contents
+                with open(file_path, encoding="utf8") as file:
+                    content = file.read().replace(
+                        "BUILDTIME_ASSETPREFIX_REPLACE_ME", f"{self._app_path_prefix}"
```

Comment:
is there no way to get this out of the js payloads and back to only being in the root html ?

Reply:
Basically Next.js has an option `assetPrefix` that gets inserted into the JS at build time. CRA builds just used the `__webpack_public_path__` global in JS instead which allowed us to set it dynamically. I could open a ticket with Next.js to see if they could let us do that but so far all of the suggestions I've found are to do something like this. 

Although I just found [this suggestion](https://github.com/twopluszero/next-images/issues/12#issuecomment-550077810). I'll see if that works, if so then I can force the nextjs builds to also rely on `__webpack_public_path__` (though it does mean we'll need to add back some of the hacky stuff where we manually threading that global through to our webworkers).